code/modules/mob/living/carbon/human/human_helpers.dm

```diff
@@ -220,3 +220,16 @@
 
 /mob/living/carbon/human/get_biological_state()
 	return dna.species.get_biological_state()
+
+///Returns death message for mob examine text
+/mob/living/carbon/human/proc/generate_death_examine_text()
+	var/mob/dead/observer/ghost = get_ghost(TRUE, TRUE)
+	if(getorgan(/obj/item/organ/brain) && !key) //The death mob has a brain and no client/player that is assigned to the mob
```

Comment:
Is key set to false when a player dies? Does converting from client to key maintain behavior?

Reply:
I was looking at the code before https://github.com/tgstation/tgstation/pull/55399 was merged and before that we used
`if(getorgan(/obj/item/organ/brain) && !key && !get_ghost(FALSE, TRUE))` to determine the "soul has departed" part of the message. Because of this I changed it back to key.
As far as I know key is used internally by Byond to know which mob should be controlled by a player when he joins the server. When you die and start to spectate as ghost the key var gets nulled because you are now controlling the ghost. This means if someone died, left the server but did not spectate the if will still be true and it will display "[t_He] [t_is] limp and unresponsive; there are no signs of life..." even though the player has disconnected. (after all the player could reconnect)
Client as far as I know determines if someone is currently controlling the mob. So if I would check for client it would jump between "[t_He] [t_is] limp and unresponsive; there are no signs of life and [t_his] soul has lost the will to live..." (there is no ghost if you never leave your body) and "[t_He] [t_is] limp and unresponsive; there are no signs of life..." depending if the play is currently disconnected or not.

Im honestly sure how many infos we want to give players about the status of someone else so I changed the messages to the format I described in the PR description.